### PR TITLE
feat: add initial sass typography API

### PR DESCRIPTION
**Pull request by crisbeto**
Adds an initial implementation for the API that'll be used to do the Material typography.

**Note:** These are just the various functions and mixins, they aren't being used anywhere yet. The usage for the `mat-typography` mixin can be:

```scss
// Includes the default typography styles.
@include mat-typography();

// Overrides one of the typography levels
@include mat-typography(
  $display-2: mat-typography-level(50px, 52px, 600)
);
```

### CODE REVIEW

File: src/lib/core/typography/_typography-utils.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Utility for fetching a nested value from a typography config.
+@function _mat-get-type-value($config, $level, $name) {
+  @return map-get(map-get($config, $level), $name);
+}
+
+// Returns the font size for a level inside a typography config.
```
Comment (jelbourn) [inline src/lib/core/typography/_typography-utils.scss]: `Returns` -> `Gets` here and elsewhere

File: src/lib/core/typography/_typography-utils.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  @return _mat-get-type-value($config, $level, font-weight);
+}
+
+// Converts a typography level into CSS styles.
+@mixin mat-typography-level-to-styles($config, $level) {
+  $font-size: mat-font-size($config, $level);
+  $font-weight: mat-font-weight($config, $level);
+  $line-height: mat-line-height($config, $level);
+
+  font: $font-weight #{$font-size}/#{$line-height} unquote(map-get($config, font-family));
```
Comment (jelbourn) [inline src/lib/core/typography/_typography-utils.scss]: Add a comment for what's going on here?

**Comment by crisbeto** [comment]
Addressed the feedback.

File: src/lib/core/typography/_typography-utils.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+// Converts a typography level into CSS styles.
+@mixin mat-typography-level-to-styles($config, $level) {
+  $font-size: mat-font-size($config, $level);
+  $font-weight: mat-font-weight($config, $level);
+  $line-height: mat-line-height($config, $level);
+  $font-family: mat-font-family($config);
+
+  // Use the shorthand `font` to represent a typography level, because it's the shortest. Notes that
+  // we need to use interpolation for `font-size/line-height` in order to prevent SASS from diving
```
Comment (willshowell) [inline src/lib/core/typography/_typography-utils.scss]: 🏊 "dividing"

**Comment by jelbourn** [comment]
LGTM, can add merge-ready when you're ready

**Comment by robertorfischer** [comment]
When are you intending to have this be usable by the other material libraries in this repo and by user code? This looks promising!

**Comment by cvgaviao** [comment]
is there any example or doc reference for this feature?

**Comment by crisbeto** [comment]
There's a guide at https://material.angular.io/guide/typography

**Comment by cvgaviao** [comment]
@crisbeto, thanks.

**Pull request merged**